MMR.Randomizer/Utils/RomUtils.cs

```diff
@@ -337,14 +337,29 @@ public static byte[] BuildROM()
                     continue;
                 }
                 RomData.MMFileList[i].Cmp_Addr = ROMAddr;
-                int file_len = RomData.MMFileList[i].Data.Length;
+                int FileLength = RomData.MMFileList[i].Data.Length;
                 if (RomData.MMFileList[i].IsCompressed)
                 {
-                    RomData.MMFileList[i].Cmp_End = ROMAddr + file_len;
+                    RomData.MMFileList[i].Cmp_End = ROMAddr + FileLength;
                 }
-                ReadWriteUtils.Arr_Insert(RomData.MMFileList[i].Data, 0, file_len, ROM, ROMAddr);
-                ROMAddr += file_len;
+                if (ROMAddr + FileLength > ROM.Length) // rom too small
+                {
+                    // assuming the largest file isn't the last one, we still want some extra space for further files
+                    //  padding will reduce the requirements for further resizes
+                    int ExpansionIncrementSize = 0x40000; // 1mb might be too large, not sure if there is a hardware compatiblity issue here
```

Comment:
I think the max rom size is 64MB? should check with someone with an everdrive if a MMR expanded rom works for them. Should limit this to whatever the compatibility limit is. Also probably worth checking WiiVC.

Reply:
We can test it but right now it's kinda theoretical. Of all the songs I've made that inject samples, I have never seen the rom expand beyond 32mb naturally, to force it past 32mb for testing I had to make a fake dummy file with a massive sample.

For now I think its probably fine to add a popup if the rom goes over 64mb that says "This rom might be too big for everdrive (or VC if testing reveals that) please remove songs or other content that increases rom size" instead of just blocking generation(emulator can handle up to 128mb at least), but we can put a full error and stop generating too.

Reply:
Added a warning if rom goes over 64mb